src/vg/civcraft/mc/civmenu/TOSListener.java

```diff
@@ -66,13 +65,27 @@ public void run() {
 	public void playerMoveEvent(PlayerMoveEvent event) {
 		Player p = event.getPlayer();
 		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
-			if(event.getTo().distance(locations.get(p.getUniqueId())) > plugin.GetConfig().get("terms.MovementRange").getInt()){
+			Location from = locations.get(p.getUniqueId());
+			if (from == null){
+				from = event.getFrom();
+				locations.put(p.getUniqueId(), from);
```

Comment:
Yeah, but the original issue was that there were unregistered players who were not bound to any location somehow (different issue than the respawning), so there were NPE thrown all the time when it tried to get their bound location. Your solution completly solves the symptoms of the issue, I was just curious whether we know what caused it in the first place (I dont think so).